Source/WebCore/Modules/webcodecs/WebCodecsVideoFrame.cpp

```diff
@@ -284,24 +284,25 @@ static std::optional<Exception> validateI420Sizes(const WebCodecsVideoFrame::Buf
 // https://w3c.github.io/webcodecs/#dom-videoframe-videoframe-data-init
 ExceptionOr<Ref<WebCodecsVideoFrame>> WebCodecsVideoFrame::create(ScriptExecutionContext& context, BufferSource&& data, BufferInit&& init)
 {
+    ASSERT(init.format);
     if (!isValidVideoFrameBufferInit(init))
         return Exception { ExceptionCode::TypeError, "buffer init is not valid"_s };
 
     DOMRectInit defaultRect { 0, 0, static_cast<double>(init.codedWidth), static_cast<double>(init.codedHeight) };
-    auto parsedRectOrExtension = parseVisibleRect(defaultRect, init.visibleRect, init.codedWidth, init.codedHeight, init.format);
+    auto parsedRectOrExtension = parseVisibleRect(defaultRect, init.visibleRect, init.codedWidth, init.codedHeight, *init.format);
```

Comment:
Can we replace this unconditional dereference with a `.value_or()`?

Reply:
This method is called from JS binding, where format should never be nullopt, hence the ASSERT and the unconditional dereference.
I'll add a local var.